packages/electron-updater/src/MacUpdater.ts

```diff
@@ -50,7 +50,21 @@ export class MacUpdater extends AppUpdater {
       log.warn(`sysctl shell command to check for macOS Rosetta environment failed: ${e}`)
     }
 
-    const isArm64Mac = process.arch === "arm64" || isRosetta
+    // const isArm64Mac = process.arch === "arm64" || isRosetta
+
+    const getArch = () => {
+      return new Promise<Boolean>((resolve) => {
+        exec('uname -a', (err: any, stdout: string) => {
+          if(err) {
+            resolve(process.arch === 'arm64')
+            return
+          }
+          resolve(stdout.includes("ARM"))
+        })
+      })
+    }
+
+    const isArm64Mac = await getArch() || isRosetta;
```

Comment:
We need a try-catch here, and I'm not sure why we can't still use `execFileSync` to mirror the other code. I think we should replace the entire section with:
```

let isArm64Mac = false

// detect if we are running inside Rosetta emulation
const sysctlRosettaInfoKey = "sysctl.proc_translated"
try {
  this.debug("Checking for macOS Rosetta environment")
  const result = execFileSync("sysctl", [sysctlRosettaInfoKey], { encoding: "utf8" })
  const isRosetta = result.includes(`${sysctlRosettaInfoKey}: 1`)
  log.info(`Checked for macOS Rosetta environment (isRosetta=${isRosetta})`)
  isArm64Mac = isRosetta
} catch (e) {
  log.warn(`sysctl shell command to check for macOS Rosetta environment failed: ${e}`)
}

try {
  this.debug("Checking for arm64 in uname")
  const result = execFileSync("uname", ['-a'], { encoding: "utf8" })
  const isArm = result.includes('ARM')
  log.info(`Checked 'uname -a': arm64=${isArm}`)
  isArm64Mac = isArm64Mac || isArm
} catch (e) {
  log.warn(`uname shell command to check for arm64 failed: ${e}`)
}

isArm64Mac = isArm64Mac || process.arch === 'arm64'
```

Reply:
appreciate！ I will improve my code with your suggestion and resolve it